api/src/main/kotlin/com/techinsights/api/service/post/PostLikeService.kt

```
@@ -0,0 +1,52 @@
+        } else {
+            try {
+                val newLike = PostLike(
+                    id = Tsid.generate().decode(),
+                    postId = postIdLong,
+                    userId = userId,
+                    ipAddress = ipAddress
+                )
+                postLikeRepository.save(newLike)
+                postRepository.incrementLikeCount(postIdLong)
+                return true
+            } catch (e: DataIntegrityViolationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)